lib/layout/list_view.dart

```diff
@@ -158,6 +160,18 @@ class ListViewState extends WidgetState<ListView> with TemplatedWidgetState {
     );
   }
 
+  // ----------------- To GET the current [index] of the item in data array -------------------
+
+  _onItemTapped(int index) {
+    if (index != widget._controller.selectedItemIndex &&
+        widget._controller.onItemTap != null) {
+      widget._controller.selectedItemIndex = index;
+      //log("Changed to index $index");
+      ScreenController().executeAction(context, widget._controller.onItemTap!,event: EnsembleEvent(widget));
```

Comment:
@Moeez1999 I am not sure if you are understanding the code. By just sending the widget as part of EnsembleEvent, you achieve nothing at all. This code change is of no use. You have to trace how events are used.. Then create your own event that extends EnsembleEvent and pass the selectedIndex and selectedItem as data on that event. Please fix this.